PrivateHospitals/backend/PrivateHospitals/PrivateHospitals.Application/Services/Appointment/AppointmentService.cs

```diff
@@ -22,80 +23,133 @@ public class AppointmentService(
+        if (doctor is null)
         {
-            return Result<bool>.ErrorResponse(new List<string> {"Doctor is not available"});
+            return Result<bool>.ErrorResponse(new List<string> { "Doctor not found" });
         }
         
-        var appointment = _mapper.Map<Core.Models.Appointment>(appointmentDto);
-        appointment.AppointmentDate = appointmentDto.Date.ToUniversalTime();
-
-        _appointmentRepository.CreateAppointmentAsync(appointment);
-
-        return Result<bool>.SuccessResponse(true);
-    }
-
-    public async Task<Result<List<AppointmentDto>>> GetAppointmentsBySpecialityAsync(string patientId, DoctorSpecialities? speciality)
-    {
-        var patient = await _patientRepository.GetPatientByIdAsync(patientId);
-        if (patient == null)
-        {
-            return Result<List<AppointmentDto>>.ErrorResponse(new List<string> {"Patient not found"});
-        }
+        var isDoctorAvailable = await _doctorRepository.IsDoctorAvailableAsync(appointmentDto.DoctorId, appointmentDto.Date);
 
-        if (speciality == null)
+        if (!isDoctorAvailable)
         {
-            return Result<List<AppointmentDto>>.ErrorResponse(new List<string> {"Doctor speciality can`t be null"});
+            return Result<bool>.ErrorResponse(new List<string> { "Doctor is not available" });
         }
 
-        var appointments = await _appointmentRepository.GetAppointmentBySpecialityAsync(patientId, speciality);
-        if (appointments.Count == 0)
+        if (appointmentDto.Date < DateTime.Now)
         {
-            return Result<List<AppointmentDto>>.ErrorResponse(new List<string> { "Appointment not found" });
+            return Result<bool>.ErrorResponse(new List<string> { "Appointment date has to be in the future" });
         }
-
-        var appointmentDto = _mapper.Map<List<AppointmentDto>>(appointments);
         
-        return Result<List<AppointmentDto>>.SuccessResponse(appointmentDto);
-    }
+        var appointment = _mapper.Map<Core.Models.Appointment>(appointmentDto);
+        appointment.AppointmentDate = appointmentDto.Date;
+        appointment.PatientId = patientId;
+        
+        await _appointmentRepository.CreateAppointmentAsync(appointment);
+        
+        return Result<bool>.SuccessResponse(true);
+     }
 
-    public async Task<Result<List<AppointmentDto>>> GetAppointmentsByDateAsync(string patientId, DateTime? fromDate, DateTime? toDate)
-    {
-        var patient = await _patientRepository.GetPatientByIdAsync(patientId);
-        if (patient == null)
-        {
-            return Result<List<AppointmentDto>>.ErrorResponse(new List<string> {"Patient not found"});
-        }
+     public async Task<Result<PaginationResult<AppointmentDto>>> GetAppointmentsAsync(AppointmentFilter appointmentFilter, string patientId)
+     {
+         if (appointmentFilter.Speciality != null)
+         {
+             if (appointmentFilter.PageNumber == null || appointmentFilter.PageSize == 0)
```

Comment:
Add default values as PageNumber = 1 & PageSize = 20. this way u avoid these identical `if`s